rdt/transformers/numerical.py

```diff
@@ -151,14 +150,10 @@ def reverse_transform(self, data):
         if self.nan is not None:
             data = self.null_transformer.reverse_transform(data)
 
-        if self._rounding_digits is not None:
-            data = data.round(self._rounding_digits)
+        if self._rounding_digits is not None or np.dtype(self._dtype).kind == 'i':
+            data = data.round(self._rounding_digits or 0)
 
-        if np.dtype(self._dtype).kind == 'i':
-            if pd.notnull(data).all():
-                return data.round().astype(self._dtype)
-
-            data[pd.notnull(data)] = np.round(data[pd.notnull(data)]).astype(self._dtype)
+        if pd.isnull(data).any():
```

Comment:
why are we returning data here?

Reply:
Because if there are nulls we cannot do the `astype` below, otherwise an `astype(np.int)` will silently replace the NaNs with the minimum available integer value.
I will add another check though, to ensure that we only enter this `if` if the dtype is actually an integer.